crates/iota-indexer/src/handlers/checkpoint_handler.rs

```diff
@@ -228,20 +228,33 @@ impl CheckpointHandler {
         let system_state: IotaSystemStateSummary =
             get_iota_system_state(&checkpoint_object_store)?.into_iota_system_state_summary();
 
-        let epoch_event = transactions
+        let event = transactions
             .iter()
             .flat_map(|t| t.events.as_ref().map(|e| &e.data))
             .flatten()
-            .find(|ev| ev.is_system_epoch_info_event())
+            .find(|ev| ev.is_system_epoch_info_event_v1() || ev.is_system_epoch_info_event_v2())
+            .map(|ev| {
+                if ev.is_system_epoch_info_event_v2() {
+                    SystemEpochInfoEvent::V2(
+                        bcs::from_bytes::<SystemEpochInfoEventV2>(&ev.contents).expect(
+                            "event deserialization should succeed as type was pre-validated",
+                        ),
+                    )
+                } else {
+                    SystemEpochInfoEvent::V1(
+                        bcs::from_bytes::<SystemEpochInfoEventV1>(&ev.contents).expect(
+                            "event deserialization should succeed as type was pre-validated",
+                        ),
+                    )
+                }
+            })
             .unwrap_or_else(|| {
                 panic!(
                     "Can't find SystemEpochInfoEventV1 in epoch end checkpoint {}",
```

Comment:
Should this be just `SystemEpochInfoEvent` as it could also be V2?
```suggestion
                    "Can't find SystemEpochInfoEvent in epoch end checkpoint {}",
```